receiver/dcgmreceiver/client.go

```diff
@@ -367,7 +371,11 @@ func (client *dcgmClient) cleanup() {
 // It returns the estimated polling interval.
 func (client *dcgmClient) collect() (time.Duration, error) {
 	client.logger.Debugf("Polling DCGM daemon for field values")
-	fieldValues, pollTime, err := dcgmGetValuesSince(dcgm.GroupAllGPUs(), client.enabledFieldGroup, client.lastSuccessfulPoll)
+	if len(client.enabledFieldIDs) == 0 {
+		// Make sure we don't try to scrape without a device group (since we don't construct one when there are no enabled fields).
```

Comment:
I wonder, should we even create a client when there are no enabled fields?

Reply:
Great question. I'll defer that to future cleanup, though, since I'm not changing it here.